tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -278,3 +286,54 @@ def extract_error(e):
     except Exception:
         return None
     return ERROR_REASON.search(edata)
+
+
+class GcpLocation(object):
+    """
+    Consists of locations applicable to
+    `KMS <https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations/list>`_ and
+    `App Engine <https://cloud.google.com/appengine/docs/admin-api/reference/rest/v1
+    /apps.locations/list>`_
+    """
+    kms_locations = ["asia",
```

Comment:
so rather duplicate long lists of names it would be nicer to make this  more like
```
locations = {
    'asia-east2': ['kms', 'appengine']
}
```
which removes the duplication, and then a helper method on this class like

```
@classmethod
get_service_regions(cls, service)
```

so the data structure is abstracted to the callers

Reply:
Rewritten this way, thank you for the advice.